database/insertData/dev/03_template_D_basicReviewTesting.js

```diff
@@ -322,6 +506,132 @@ exports.queries = [
+                  }
+                  serial: {
+                    operator: "objectProperties"
+                    children: ["outputCumulative.generatedText"]
+                  }
+#                   application_id: {
+#                     operator: "objectProperties"
+#                     children: ["applicationData.applicationId"]
+#                   }
+                }
+              }
+              {
+                actionCode: "scheduleAction"
+                trigger: ON_REVIEW_SUBMIT
+                sequence: 102
+                condition: {
+                  operator: "="
+                  children: [
+                    {
+                      operator: "objectProperties"
+                      children: ["applicationData.outcome"]
+                    }
+                    "APPROVED"
+                  ]
+                }
+                parameterQueries:{
+                  eventCode: "prod_exp1"
+                  duration: { minute: 5 }
+                }
+              }
+              {
+                actionCode: "sendNotification"
+                trigger: ON_SCHEDULE
+                eventCode: "prod_exp1"
+                condition: {
+                  operator: "=",
+                  children: [
+                    {
+                      operator: "graphQL",
+                      children: [
+                        "query CheckExpired($id: Int!) {products(filter: {id: {equalTo: $id}, expiryDate: {lessThan: \\"now()\\"}}) {totalCount}}",
+                        "graphQLEndpoint",
+                        [
+                          "id"
+                        ],
+                        {
+                          operator: "objectProperties",
+                          children: [
+                            "outputCumulative.product.id"
+                          ]
+                        },
+                        "products.totalCount"
+                      ]
+                    },
+                    1
+                  ]
+                }
+                parameterQueries: {
+                  fromName: "Application Manager"
+                  fromEmail: "no-reply@sussol.net"
+                  email: {
+                    operator: "objectProperties"
+                    children: ["applicationData.email", ""]
+                  }
+                  subject: {
+                    operator: "stringSubstitution"
+                    children: [
+                      "Your product %1 has expired",
+                      {
+                        operator: "objectProperties"
+                        children: [ "outputCumulative.product.name", ""]
+                      }
+                    ]
+                  }
+                  message: "Your product registration has expired. If you wish to continue, you'll need to re-apply.\\n(Note: this is just a test to check the notification worked. Nothing proper has actually changed)"
+                }
+              }
```

Comment:
It could be useful to have the "checkExpiry" query to be internally defined somehow, since we'd be doing it a lot and it is quite complicated. 

There's complications to all ways of doing things, though, so it'd good to maybe have a chat about what would actually help and how to achieve it.